Sources/Moya/Moya+Alamofire.swift

```diff
@@ -75,16 +76,45 @@ internal protocol Requestable {
 
 extension DataRequest: Requestable {
     internal func response(callbackQueue: DispatchQueue?, completionHandler: @escaping RequestableCompletion) -> Self {
-        return response(queue: callbackQueue) { handler  in
-            completionHandler(handler.response, handler.request, handler.data, handler.error)
+        if let callbackQueue = callbackQueue {
+            return response(queue: callbackQueue) { handler  in
+                completionHandler(handler.response, handler.request, handler.data, handler.error)
+            }
+        } else {
+            return response { handler  in
+                completionHandler(handler.response, handler.request, handler.data, handler.error)
+            }
         }
     }
 }
 
 extension DownloadRequest: Requestable {
     internal func response(callbackQueue: DispatchQueue?, completionHandler: @escaping RequestableCompletion) -> Self {
-        return response(queue: callbackQueue) { handler  in
-            completionHandler(handler.response, handler.request, nil, handler.error)
+        if let callbackQueue = callbackQueue {
```

Comment:
Not related, but we are doing this `if let callBack` twice (here and on DataRequest), should we move this logic to `response(queue:` so we could just pass the callback and it could be handled there?

Reply:
Ah yes it would be but the `response(queue:` is an Alamofire func and these two funcs are just helpers that enable queue as an optional queue instead of this if/else 😄